server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -170,6 +176,79 @@ public List<ScimUser> retrieveByUsernameAndZone(String username, String zoneId)
         return jdbcTemplate.query(USER_BY_USERNAME_AND_ZONE_QUERY , mapper, username, zoneId);
     }
 
+    /**
+     * This method should behave similar to {@link JdbcScimUserProvisioning#query(String, String, boolean, String)},
+     * with the filtering for active IdPs (if required) being the only difference.
+     */
+    @Override
+    public List<ScimUser> retrieveByScimFilter(
+            final String filter,
+            final boolean includeInactive,
+            final String sortBy,
```

Comment:
Is there a better type than String that we can use for sortBy so it's better constrained to valid values?

Reply:
The method allows passing any of the fields of the `ScimUser` class for this parameter (except for `password` and `salt`). We can introduce an enum for all the options, but I would do this in another issue/PR. What do you think?

Reply:
Yeah, it looks like other existing code is already depending on this free string, so sure, let's save that improvement for another PR.